libs/books/data-access/src/lib/+state/reading-list.effects.ts

```diff
@@ -27,41 +29,75 @@ export class ReadingListEffects implements OnInitEffects {
   addBook$ = createEffect(() =>
     this.actions$.pipe(
       ofType(ReadingListActions.addToReadingList),
-      concatMap(({ book }) =>
-        this.http.post('/api/reading-list', book).pipe(
+      concatMap(({ book, showSnackBar }) => {
+        if (showSnackBar === true) {
+          this.openSnackBar(SnackBarConstants.ADD_MESSAGE).onAction()
+          .subscribe(() => {
+            const { id, ...rest } = book;
+            this.store.dispatch(
```

Comment:
Should never call `store.dispatch` from effects class. Can you handle the undo case by creating a new effect rather than modify existing one?